Quarkus 1.12.1.Final：Build failure  in native mode  on windows

**Describe the bug**
(Describe the problem clearly and concisely.)

Build failure  in native mode  on windows with  1.12.1.Final.
It work fine with 1.12.0.Final.

```
[INFO] --- quarkus-maven-plugin:1.12.1.Final:build (default) @ code-with-quarkus ---
[INFO] [org.jboss.threads] JBoss Threads version 3.2.0.Final
[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Uber JAR strategy is used for native image source JAR generation on Windows. This is done for the time being to work around a current GraalVM limitation on Windows concerning the maximum command length (see https://github.com/oracle/graal/issues/2387).
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building fat jar: C:\temp\code-with-quarkus\target\code-with-quarkus-1.0.0-SNAPSHOT-native-image-source-jar\code-with-quarkus-1.0.0-SNAPSHOT-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Building native image from C:\temp\code-with-quarkus\target\code-with-quarkus-1.0.0-SNAPSHOT-native-image-source-jar\code-with-quarkus-1.0.0-SNAPSHOT-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on GraalVM Version 21.0.0.2 (Java Version 11.0.10+8-jvmci-21.0-b06)
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Duser.language=zh -J-Duser.country=CN -J-Dfile.encoding=UTF-8 --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy\$BySpaceAndTime -H:+JNI -H:+AllowFoldMethods -jar code-with-quarkus-1.0.0-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:EnableURLProtocols=http -H:-UseServiceLoaderFeature -H:+StackTrace code-with-quarkus-1.0.0-SNAPSHOT-runner.exe
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]    classlist:   3,359.26 ms,  0.96 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]        (cap):   2,308.23 ms,  0.96 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]        setup:   4,709.54 ms,  0.96 GB
09:06:49,719 INFO  [org.jbo.threads] JBoss Threads version 3.2.0.Final
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]     (clinit):     594.40 ms,  2.64 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]   (typeflow):  17,621.32 ms,  2.64 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]    (objects):  13,207.32 ms,  2.64 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]   (features):     494.51 ms,  2.64 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]     analysis:  32,857.40 ms,  2.64 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]     universe:   1,654.41 ms,  2.64 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]      (parse):   5,398.31 ms,  2.64 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]     (inline):   3,204.22 ms,  3.78 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]    (compile):  29,779.77 ms,  4.88 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]      compile:  43,271.05 ms,  4.88 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]        image:   3,742.70 ms,  4.88 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]        write:     730.29 ms,  4.88 GB
[code-with-quarkus-1.0.0-SNAPSHOT-runner.exe:11972]      [total]:  90,685.81 ms,  4.88 GB
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:54 min
[INFO] Finished at: 2021-03-04T09:07:57+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.12.1.Final:build (default) on project code-with-quarkus: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to build native image
[ERROR]         at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:281)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: java.nio.file.NoSuchFileException: C:\temp\code-with-quarkus\target\code-with-quarkus-1.0.0-SNAPSHOT-native-image-source-jar\code-with-quarkus-1.0.0-SNAPSHOT-runner.exe
[ERROR]         at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
[ERROR]         at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
[ERROR]         at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
[ERROR]         at java.base/sun.nio.fs.WindowsFileAttributeViews$Basic.readAttributes(WindowsFileAttributeViews.java:53)
[ERROR]         at java.base/sun.nio.fs.WindowsFileAttributeViews$Basic.readAttributes(WindowsFileAttributeViews.java:38)
[ERROR]         at java.base/sun.nio.fs.WindowsFileSystemProvider.readAttributes(WindowsFileSystemProvider.java:198)
[ERROR]         at java.base/java.nio.file.Files.readAttributes(Files.java:1764)
[ERROR]         at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:225)
[ERROR]         at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276)
[ERROR]         at java.base/java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:322)
[ERROR]         at java.base/java.nio.file.Files.walkFileTree(Files.java:2717)
[ERROR]         at io.quarkus.bootstrap.util.IoUtils.copy(IoUtils.java:126)
[ERROR]         at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:257)
[ERROR]         ... 10 more
[ERROR] -> [Help 1]
```


**Expected behavior**
(Describe the expected behavior clearly and concisely.)

**Actual behavior**
(Describe the actual behavior clearly and concisely.)


**To Reproduce**


Steps to reproduce the behavior:
1.  gennate a sample project with https://code.quarkus.io/
2.  mvn clean  package -Pnative
3. 

**Configuration**
```properties
# Add your application.properties here, if applicable.

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`:  windows  10.0.19041.804
 - Output of `java -version`:   build 11.0.7+8-LTS
 - GraalVM version (if different from Java):  graalvm-ce-java11-21.0.0.2
 - Quarkus version or git rev:   1.12.1.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 

**Additional context**
(Add any other context about the problem here.)

--- a/.github/workflows/ci-actions-incremental.yml
+++ b/.github/workflows/ci-actions-incremental.yml
@@ -34,6 +34,7 @@ env:
   # Workaround testsuite locale issue
   LANG: en_US.UTF-8
   NATIVE_TEST_MAVEN_OPTS: "-B --settings .github/mvn-settings.xml --fail-at-end -Dquarkus.native.container-build=true -Dtest-containers -Dstart-containers -Dnative-image.xmx=5g -Dnative -Dnative.surefire.skip -Dformat.skip -Dno-descriptor-tests install -DskipDocs"
+  WINDOWS_NATIVE_TEST_MAVEN_OPTS: "-B --settings .github/mvn-settings.xml --fail-at-end -Dtest-containers -Dstart-containers -Dnative-image.xmx=5g -Dnative -Dnative.surefire.skip -Dformat.skip -Dno-descriptor-tests install -DskipDocs"
   JVM_TEST_MAVEN_OPTS: "-e -B --settings .github/mvn-settings.xml -Dtest-containers -Dstart-containers -Dformat.skip -DskipDocs"
   DB_USER: hibernate_orm_test
   DB_PASSWORD: hibernate_orm_test
@@ -653,4 +654,80 @@ jobs:
         if: failure()
         with:
           name: test-reports-native-${{matrix.category}}
+          path: 'test-reports.tgz'
+  native-tests-windows:
+    name: Native Tests - Windows - ${{matrix.category}}
+    needs: [build-jdk11, native-tests-read-json-matrix]
+    runs-on: windows-latest
+    # Skip master in forks
+    if: "github.repository == 'quarkusio/quarkus' || !endsWith(github.ref, '/master')"
+    # Ignore the following YAML Schema error
+    timeout-minutes: ${{matrix.timeout}}
+    strategy:
+      max-parallel: 8
+      fail-fast: false
+      # we only run a native build on Windows for Hibernate Validator just to be sure
+      # that we can build native executables on Windows
+      # matrix: ${{ fromJson(needs.native-tests-read-json-matrix.outputs.matrix) }}
+      matrix:
+        category: [hibernate-validator]
+        include:
+          - category: hibernate-validator
+            timeout: 20
+            test-modules: hibernate-validator
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Add quarkusio remote
+        shell: bash
+        run: git remote add quarkusio https://github.com/quarkusio/quarkus.git
+      - name: Set up JDK 11
+        # Uses sha for added security since tags can be updated
+        uses: joschi/setup-jdk@e87a7cec853d2dd7066adf837fe12bf0f3d45e52
+        with:
+          java-version: 11
+      - name: Install cl.exe
+        uses: ilammy/msvc-dev-cmd@v1
+      - name: Setup GraalVM
+        id: setup-graalvm
+        uses: DeLaGuardo/setup-graalvm@master
+        with:
+          graalvm-version: '21.0.0.java11'
+      - name: Install native-image component
+        run: |
+          gu.cmd install native-image
+      - name: Reclaim Disk Space
+        run: .github/ci-prerequisites.sh
+      # We do this so we can get better analytics for the downloaded version of the build images
+      - name: Update Docker Client User Agent
+        shell: bash
+        run: |
+          cat <<< $(jq '.HttpHeaders += {"User-Agent": "Quarkus-CI-Docker-Client"}' ~/.docker/config.json) > ~/.docker/config.json
+      - name: Download Maven Repo
+        uses: actions/download-artifact@v1
+        with:
+          name: maven-repo
+          path: .
+      - name: Extract Maven Repo
+        shell: bash
+        run: tar -xzf maven-repo.tgz -C ~
+      - name: Build with Maven
+        shell: bash
+        env:
+          TEST_MODULES: ${{matrix.test-modules}}
+          CATEGORY: ${{matrix.category}}
+        run: |
+          for i in $TEST_MODULES
+          do modules+="integration-tests/$i,"; done
+          mvn -pl "${modules}" $WINDOWS_NATIVE_TEST_MAVEN_OPTS ${{ needs.build-jdk11.outputs.gib_args }}
+      - name: Prepare failure archive (if maven failed)
+        if: failure()
+        shell: bash
+        run: find . -type d -name '*-reports' -o -wholename '*/build/reports/tests/functionalTest' -o -name '*.log' | tar -czf test-reports.tgz -T -
+      - name: Upload failure Archive (if maven failed)
+        uses: actions/upload-artifact@v1
+        if: failure()
+        with:
+          name: test-reports-native-windows-${{matrix.category}}
           path: 'test-reports.tgz'

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java
@@ -14,12 +14,12 @@ public class NativeImageBuildRemoteContainerRunner extends NativeImageBuildConta
 
     private static final Logger log = Logger.getLogger(NativeImageBuildRemoteContainerRunner.class);
 
-    private final String nativeImageName;
+    private final String resultingExecutableName;
     private String containerId;
 
-    public NativeImageBuildRemoteContainerRunner(NativeConfig nativeConfig, Path outputDir, String nativeImageName) {
+    public NativeImageBuildRemoteContainerRunner(NativeConfig nativeConfig, Path outputDir, String resultingExecutableName) {
         super(nativeConfig, outputDir);
-        this.nativeImageName = nativeImageName;
+        this.resultingExecutableName = resultingExecutableName;
     }
 
     @Override
@@ -47,7 +47,7 @@ public class NativeImageBuildRemoteContainerRunner extends NativeImageBuildConta
 
     @Override
     protected void postBuild() throws InterruptedException, IOException {
-        copyFromBuilder(nativeImageName, "Failed to copy native image from container back to the host.");
+        copyFromBuilder(resultingExecutableName, "Failed to copy native executable from container back to the host.");
         if (nativeConfig.debug.enabled) {
             copyFromBuilder("sources", "Failed to copy sources from container back to the host.");
         }

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
@@ -89,7 +89,7 @@ public class NativeImageBuildStep {
 
         Path runnerJar = outputDir.resolve(nativeImageSourceJarBuildItem.getPath().getFileName());
 
-        String nativeImageName = getResultingBinaryName(outputTargetBuildItem, packageConfig, isContainerBuild(nativeConfig));
+        String nativeImageName = getNativeImageName(outputTargetBuildItem, packageConfig);
 
         NativeImageInvokerInfo nativeImageArgs = new NativeImageInvokerInfo.Builder()
                 .setNativeConfig(nativeConfig)
@@ -98,7 +98,7 @@ public class NativeImageBuildStep {
                 .setOutputDir(outputDir)
                 .setRunnerJarName(runnerJar.getFileName().toString())
                 // the path to native-image is not known now, it is only known at the time the native-sources will be consumed
-                .setResultingBinaryName(nativeImageName)
+                .setNativeImageName(nativeImageName)
                 .setContainerBuild(nativeConfig.containerRuntime.isPresent() || nativeConfig.containerBuild)
                 .build();
         List<String> command = nativeImageArgs.getArgs();
@@ -144,9 +144,10 @@ public class NativeImageBuildStep {
             noPIE = detectNoPIE();
         }
 
-        String nativeImageName = getResultingBinaryName(outputTargetBuildItem, packageConfig, isContainerBuild);
+        String nativeImageName = getNativeImageName(outputTargetBuildItem, packageConfig);
+        String resultingExecutableName = getResultingExecutableName(nativeImageName, isContainerBuild);
 
-        NativeImageBuildRunner buildRunner = getNativeImageBuildRunner(nativeConfig, outputDir, nativeImageName);
+        NativeImageBuildRunner buildRunner = getNativeImageBuildRunner(nativeConfig, outputDir, resultingExecutableName);
         buildRunner.setup(processInheritIODisabled.isPresent());
         final GraalVM.Version graalVMVersion = buildRunner.getGraalVMVersion();
 
@@ -167,7 +168,7 @@ public class NativeImageBuildStep {
                     .setNativeImageProperties(nativeImageProperties)
                     .setOutputDir(outputDir)
                     .setRunnerJarName(runnerJarName)
-                    .setResultingBinaryName(nativeImageName)
+                    .setNativeImageName(nativeImageName)
                     .setNoPIE(noPIE)
                     .setContainerBuild(isContainerBuild)
                     .setGraalVMVersion(graalVMVersion)
@@ -179,10 +180,10 @@ public class NativeImageBuildStep {
             if (exitCode != 0) {
                 throw imageGenerationFailed(exitCode, nativeImageArgs);
             }
-            Path generatedImage = outputDir.resolve(nativeImageName);
-            Path finalPath = outputTargetBuildItem.getOutputDirectory().resolve(nativeImageName);
-            IoUtils.copy(generatedImage, finalPath);
-            Files.delete(generatedImage);
+            Path generatedExecutablePath = outputDir.resolve(resultingExecutableName);
+            Path finalExecutablePath = outputTargetBuildItem.getOutputDirectory().resolve(resultingExecutableName);
+            IoUtils.copy(generatedExecutablePath, finalExecutablePath);
+            Files.delete(generatedExecutablePath);
             if (nativeConfig.debug.enabled) {
                 final String sources = "sources";
                 final Path generatedSources = outputDir.resolve(sources);
@@ -190,20 +191,20 @@ public class NativeImageBuildStep {
                 IoUtils.copy(generatedSources, finalSources);
                 IoUtils.recursiveDelete(generatedSources);
             }
-            System.setProperty("native.image.path", finalPath.toAbsolutePath().toString());
+            System.setProperty("native.image.path", finalExecutablePath.toAbsolutePath().toString());
 
             if (objcopyExists()) {
                 if (nativeConfig.debug.enabled) {
-                    splitDebugSymbols(finalPath);
+                    splitDebugSymbols(finalExecutablePath);
                 }
                 // Strip debug symbols regardless, because the underlying JDK might contain them
-                objcopy("--strip-debug", finalPath.toString());
+                objcopy("--strip-debug", finalExecutablePath.toString());
             } else {
                 log.warn("objcopy executable not found in PATH. Debug symbols will not be separated from executable.");
                 log.warn("That will result in a larger native image with debug symbols embedded in it.");
             }
 
-            return new NativeImageBuildItem(finalPath);
+            return new NativeImageBuildItem(finalExecutablePath);
         } catch (Exception e) {
             throw new RuntimeException("Failed to build native image", e);
         } finally {
@@ -214,14 +215,17 @@ public class NativeImageBuildStep {
         }
     }
 
-    private String getResultingBinaryName(OutputTargetBuildItem outputTargetBuildItem, PackageConfig packageConfig,
-            boolean isContainerBuild) {
-        String nativeImageName = outputTargetBuildItem.getBaseName() + packageConfig.runnerSuffix;
+    private String getNativeImageName(OutputTargetBuildItem outputTargetBuildItem, PackageConfig packageConfig) {
+        return outputTargetBuildItem.getBaseName() + packageConfig.runnerSuffix;
+    }
+
+    private String getResultingExecutableName(String nativeImageName, boolean isContainerBuild) {
+        String resultingExecutableName = nativeImageName;
         if (SystemUtils.IS_OS_WINDOWS && !isContainerBuild) {
             //once image is generated it gets added .exe on Windows
-            nativeImageName = nativeImageName + ".exe";
+            resultingExecutableName = resultingExecutableName + ".exe";
         }
-        return nativeImageName;
+        return resultingExecutableName;
     }
 
     public static boolean isContainerBuild(NativeConfig nativeConfig) {
@@ -229,7 +233,7 @@ public class NativeImageBuildStep {
     }
 
     private static NativeImageBuildRunner getNativeImageBuildRunner(NativeConfig nativeConfig, Path outputDir,
-            String nativeImageName) {
+            String resultingExecutableName) {
         if (!isContainerBuild(nativeConfig)) {
             NativeImageBuildLocalRunner localRunner = getNativeImageBuildLocalRunner(nativeConfig);
             if (localRunner != null) {
@@ -244,7 +248,7 @@ public class NativeImageBuildStep {
             log.warn(errorMessage + " Attempting to fall back to container build.");
         }
         if (nativeConfig.remoteContainerBuild) {
-            return new NativeImageBuildRemoteContainerRunner(nativeConfig, outputDir, nativeImageName);
+            return new NativeImageBuildRemoteContainerRunner(nativeConfig, outputDir, resultingExecutableName);
         }
         return new NativeImageBuildLocalContainerRunner(nativeConfig, outputDir);
     }
@@ -611,7 +615,7 @@ public class NativeImageBuildStep {
             private String noPIE = "";
             private boolean isContainerBuild = false;
             private GraalVM.Version graalVMVersion = GraalVM.Version.UNVERSIONED;
-            private String resultingBinaryName;
+            private String nativeImageName;
 
             public Builder setNativeConfig(NativeConfig nativeConfig) {
                 this.nativeConfig = nativeConfig;
@@ -653,8 +657,8 @@ public class NativeImageBuildStep {
                 return this;
             }
 
-            public Builder setResultingBinaryName(String resultingBinaryName) {
-                this.resultingBinaryName = resultingBinaryName;
+            public Builder setNativeImageName(String nativeImageName) {
+                this.nativeImageName = nativeImageName;
                 return this;
             }
 
@@ -797,7 +801,7 @@ public class NativeImageBuildStep {
                     nativeImageArgs.add("-H:+DashboardAll");
                 }
 
-                nativeImageArgs.add(resultingBinaryName);
+                nativeImageArgs.add(nativeImageName);
 
                 return new NativeImageInvokerInfo(nativeImageArgs);
             }

